torch/csrc/autograd/python_function.cpp

```diff
@@ -310,38 +301,31 @@ using t2var_type = std::unordered_map<PyObject *, THPVariable *>;
 
 // Bump the counters of all recorded dirty input tensors, adding each of them
 // into dirty_inputs.  Also does some sanity checking.
-static void _mark_dirty(THPFunction *self, t2var_type &t2var,
-        std::unordered_set<PyObject *> &dirty_inputs)
+static std::unordered_set<PyObject*> _parse_dirty_inputs(THPFunction *self)
 {
   // Increase versions of modified tensors
-  if (!self->dirty_tensors) return;
+  std::unordered_set<PyObject*> dirty_inputs;
```

Comment:
NB it's probably faster to use a vector for this purpose. I don't think anyone uses function that mutate a lot of inputs